zap/src/main/java/org/zaproxy/zap/model/Tech.java

```diff
@@ -287,4 +288,28 @@ public static Tech get(String name) {
         }
         return null;
     }
+
+    /**
+     * Gets the Techs that match the name or for which the parent matches the name, or an empty Set.
+     *
+     * @param name the name of the Tech
+     * @return the matching Techs
+     * @since 2.14.0
+     */
+    public static Set<Tech> getTechs(Tech t) {
+        Set<Tech> set = new HashSet<>();
+        for (Tech tech : getAll()) {
+            if (tech.getParent() == null) {
+                continue;
+            }
+            // 3 layers, prioritized
+            if (tech.getParent().equals(t)
+                    || tech.equals(t)
+                    || (tech.getParent().getParent() != null
+                            && tech.getParent().getParent().equals(t))) {
```

Comment:
This does not cover deeper levels that we might add in the future.

Reply:
Agreed, hence the comment. I guess I could move that to the method Javadoc?

Reply:
IMO this is a bug, we should not need to change the method for newer techs (which can be added dynamically by any add-on).

Reply:
Okay, I'll look at handling it differently then.

Reply:
I came up with a better mechanism. I just need to add tests.